pipeline-steps/deploy.groovy

```diff
@@ -35,4 +35,62 @@ def deploy_sh(Map args) {
   ) // conditionalStage
 }
 
+def deploy_maas() {
+  common.conditionalStage(
+    stage_name: 'Prepare MaaS Configs',
+    stage: {
+      withCredentials([
+        string(
+          credentialsId: "dev_pubcloud_username",
+          variable: "PUBCLOUD_USERNAME"
+        ),
+        string(
+          credentialsId: "dev_pubcloud_api_key",
+          variable: "PUBCLOUD_API_KEY"
+        ),
+        string(
+          credentialsId: "dev_pubcloud_tenant_id",
+          variable: "PUBCLOUD_TENANT_ID"
+        )
+      ]){
+        sh """
+          set +x
+          export ADMIN_PWD=\$(sudo grep keystone_auth_admin_password /etc/openstack_deploy/user_osa_secrets.yml  | awk '{print \$2}')
+          set -x
+          export HOSTNAME=\$(hostname)
+
+          sudo -E sh -c 'echo "
+rackspace_cloud_username: ${env.PUBCLOUD_USERNAME}
+rackspace_cloud_api_key: ${env.PUBCLOUD_API_KEY}
+rackspace_cloud_tenant_id: ${env.PUBCLOUD_TENANT_ID}
+maas_notification_plan: npTechnicalContactsEmail
+maas_fqdn_extension: .${env.NODE_NAME}
+memory_used_percentage_warning_threshold: 99.0
+memory_used_percentage_critical_threshold: 99.5
+net_max_speed: 1000
+lb_name: \${HOSTNAME}" >> /etc/openstack_deploy/user_rpco_variables_overrides.yml'
+        """
+       } //withCredentials
+    } //stage
+  ) //conditionalStage
+  common.conditionalStage(
```

Comment:
Fair enough. I remember thinking I should really move verify-maas out of deploy as it is a test not a deploy step. However the deploy task generates a stage per playbook, so I wouldn't gain much from moving it. 

Calling the deploy stage twice would be complicated as you'd need two playbooks variables and two stage names. Easier to not use it. 

@jpmontez maybe setup maas and verify maas could be functions, then I could reuse them in rpc-aio.